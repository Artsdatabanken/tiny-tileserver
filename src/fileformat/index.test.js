const format = require("./index");
const mbTileReader = require("./mbtiles/mbtileReader");

test("typeFromFileExt mbtiles", () => {
  const actual = format.getTypeFromFileExt("mbtiles");
  expect(actual).toMatchSnapshot();
});

test("typeFromFileExt sqlite", () => {
  const actual = format.getTypeFromFileExt("sqlite");
  expect(actual).toMatchSnapshot();
});

test("typeFromFileExt png", () => {
  const actual = format.getTypeFromFileExt("png");
  expect(actual).toMatchSnapshot();
});

const cursor = {
  type: "mbtiles",
  physicalDir: "./testdata/pbf.mbtiles",
  pathSegments: [],
  query: {},
  content: { format: "pbf" },
  link: ""
};

test("get z", async () => {
  const actual = await load(cursor, []);
  expect(actual).toMatchSnapshot();
});
test("get zx", async () => {
  const actual = await load(cursor, [3]);
  expect(actual).toMatchSnapshot();
});
test("get zxy", async () => {
  const actual = await load(cursor, [3, 4]);
  expect(actual).toMatchSnapshot();
});
test("get tile", async () => {
  const actual = await load(cursor, [3, 4, 2]);
  expect(actual).toMatchSnapshot();
});

async function load(cursor, path) {
  cursor = { ...cursor, pathSegments: path };
  await format.load(cursor);
  return cursor;
}
