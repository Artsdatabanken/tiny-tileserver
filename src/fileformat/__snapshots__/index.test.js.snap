// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get tile 1`] = `
Object {
  "buffer": Object {
    "data": Array [
      31,
      139,
      8,
      0,
      0,
      0,
      0,
      0,
      2,
      3,
      147,
      138,
      173,
      96,
      226,
      226,
      112,
      244,
      143,
      55,
      52,
      212,
      53,
      49,
      213,
      104,
      80,
      144,
      98,
      73,
      44,
      74,
      77,
      148,
      98,
      73,
      206,
      79,
      73,
      149,
      98,
      201,
      75,
      204,
      77,
      85,
      226,
      230,
      226,
      52,
      180,
      176,
      180,
      48,
      182,
      52,
      50,
      49,
      83,
      226,
      66,
      168,
      86,
      98,
      227,
      98,
      113,
      202,
      207,
      206,
      19,
      146,
      148,
      96,
      22,
      98,
      99,
      96,
      96,
      100,
      100,
      98,
      82,
      226,
      229,
      124,
      196,
      126,
      71,
      76,
      74,
      65,
      194,
      80,
      137,
      89,
      147,
      31,
      0,
      216,
      169,
      196,
      67,
      95,
      0,
      0,
      0,
    ],
    "type": "Buffer",
  },
  "content": Object {
    "format": "pbf",
  },
  "contentType": "application/x-protobuf",
  "link": "",
  "pathSegments": Array [
    3,
    4,
    2,
  ],
  "physicalDir": "./testdata/pbf.mbtiles",
  "query": Object {},
  "type": "mbtiles",
}
`;

exports[`get z 1`] = `
Object {
  "content": Object {
    "format": "pbf",
  },
  "link": "",
  "pathSegments": Array [],
  "physicalDir": "./testdata/pbf.mbtiles",
  "query": Object {},
  "type": "mbtiles",
}
`;

exports[`get zx 1`] = `
Object {
  "canBrowse": true,
  "content": Object {
    "format": "pbf",
  },
  "files": Array [
    Object {
      "filesize": undefined,
      "name": "4",
    },
  ],
  "link": "",
  "pathSegments": Array [
    3,
  ],
  "physicalDir": "./testdata/pbf.mbtiles",
  "query": Object {},
  "type": "mbtiles",
}
`;

exports[`get zxy 1`] = `
Object {
  "canBrowse": true,
  "content": Object {
    "format": "pbf",
  },
  "files": Array [
    Object {
      "alternateFormats": Object {
        "geojson": "2?format=geojson",
        "pbfjson": "2?format=pbfjson",
        "png": "2?format=png&border=rgba(128,128,128,0.2)&size=768&font=8px%20Tahoma&fontColor=rgba(0,0,0,0.3)&stroke=rgba(0,0,0,0.4)&antialias=none",
      },
      "filesize": 107,
      "name": "2",
    },
  ],
  "link": "",
  "pathSegments": Array [
    3,
    4,
  ],
  "physicalDir": "./testdata/pbf.mbtiles",
  "query": Object {},
  "type": "mbtiles",
}
`;

exports[`typeFromFileExt mbtiles 1`] = `"mbtiles"`;

exports[`typeFromFileExt png 1`] = `"file"`;

exports[`typeFromFileExt sqlite 1`] = `"sqlite"`;
